### PR TITLE
[CHORE] close actor systems in tests

### PR DESCRIPTION
**Pull request by lemastero**
# Description

Some actor systems were not closed in tests. Problem mantifest when you start `sbt` console and then run `test`. Event though tests finished Akka was still producing traffic:

```scala
[info] ScalaTest
[info] Run completed in 2 minutes, 35 seconds.
[info] Total number of tests run: 1650
[info] Suites: completed 158, aborted 0
[info] Tests: succeeded 1650, failed 0, canceled 0, ignored 2, pending 0
[info] All tests passed.
[info] Passed: Total 1650, Failed 0, Errors 0, Passed 1650, Ignored 2
19:55:08.365 ERROR i.i.e.b.sync.regular.BlockFetcher akka://BlockFetcherSpec_System/user/$g - Unexpected error while doing a request
akka.pattern.AskTimeoutException: Ask timed out on [Actor[akka://BlockFetcherSpec_System/system/testProbe-860#-2082256884]] after [3000 ms]. Message of type [io.iohk.ethereum.blockchain.sync.PeersClient$Request] was sent by [Actor[akka://BlockFetcherSpec_System/user/$g#1560748993]]. A typical reason for `AskTimeoutException` is that the recipient actor didn't send a reply.
[success] Total time: 197 s, completed Dec 8, 2020 7:55:09 PM
sbt:mantis> 19:55:12.755 ERROR i.i.e.b.sync.regular.BlockFetcher akka://BlockFetcherSpec_System/user/$g - Unexpected error while doing a request
akka.pattern.AskTimeoutException: Ask timed out on [Actor[akka://BlockFetcherSpec_System/system/testProbe-860#-2082256884]] after [3000 ms]. Message of type [io.iohk.ethereum.blockchain.sync.PeersClient$Request] was sent by [Actor[akka://BlockFetcherSpec_System/user/$g#1560748993]]. A typical reason for `AskTimeoutException` is that the recipient actor didn't send a reply.
19:55:15.775 ERROR i.i.e.b.sync.regular.BlockFetcher akka://BlockFetcherSpec_System/user/$g - Unexpected error while doing a request
akka.pattern.AskTimeoutException: Ask timed out on [Actor[akka://BlockFetcherSpec_System/system/testProbe-860#-2082256884]] after [3000 ms]. Message of type [io.iohk.ethereum.blockchain.sync.PeersClient$Request] was sent by [Actor[akka://BlockFetcherSpec_System/user/$g#1560748993]]. A typical reason for `AskTimeoutException` is that the recipient actor didn't send a reply.
19:55:18.795 ERROR i.i.e.b.sync.regular.BlockFetcher akka://BlockFetcherSpec_System/user/$g - Unexpected error while doing a request
akka.pattern.AskTimeoutException: Ask timed out on [Actor[akka://BlockFetcherSpec_System/system/testProbe-860#-2082256884]] after [3000 ms]. Message of type [io.iohk.ethereum.blockchain.sync.PeersClient$Request] was sent by [Actor[akka://BlockFetcherSpec_System/user/$g#1560748993]]. A typical reason for `AskTimeoutException` is that the recipient actor didn't send a reply.
19:55:21.815 ERROR i.i.e.b.sync.regular.BlockFetcher akka://BlockFetcherSpec_System/user/$g - Unexpected error while doing a request
akka.pattern.AskTimeoutException: Ask timed out on [Actor[akka://BlockFetcherSpec_System/system/testProbe-860#-2082256884]] after [3000 ms]. Message of type [io.iohk.ethereum.blockchain.sync.PeersClient$Request] was sent by [Actor[akka://BlockFetcherSpec_System/user/$g#1560748993]]. A typical reason for `AskTimeoutException` is that the recipient actor didn't send a reply.
```

This was found by @enriquerodbe

### CODE REVIEW

File: src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherStateSpec.scala [hunk 1]
```diff
@@ -3,14 +3,18 @@ package io.iohk.ethereum.blockchain.sync.regular
...
+import io.iohk.ethereum.{BlockHelpers, WithActorSystemShutDown}
 import io.iohk.ethereum.network.PeerId
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpecLike
 
 import scala.collection.immutable.Queue
 
-class BlockFetcherStateSpec extends TestKit(ActorSystem()) with AnyWordSpecLike with Matchers {
+class BlockFetcherStateSpec
+    extends TestKit(ActorSystem())
```
Comment (mmrozek) [inline src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherStateSpec.scala]: To consistency with other specs we should add here an actor system name - `BlockFetcherStateSpec_System`

**Review by mmrozek — APPROVED** [review]
Minor comment only. LGTM

**Review by biandratti — APPROVED** [review]

**Comment by biandratti** [comment]
LGTM!

File: src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherStateSpec.scala (hunk 1 above)
Comment (lemastero) [reply src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherStateSpec.scala]: Good catch! Fixed: https://github.com/input-output-hk/mantis/pull/845/commits/58722643fbb62009c4d3b8400168830c888636c5

**Pull request merged**